### PR TITLE
Replaced mocha with jest

### PR DESCRIPTION
**Pull request by valishah**
Made the following changes as an initial step of refactoring and optimizing the code flow

- Replaced **mocha** with **jest** and changed the syntax where it's needed. 
- Added **.editorconfig**
- Seperated **CHANGELOG.md**
- Created `__tests__` folders
- Upgraded **node** support to **>=8**

Here is how our test reports will look from now with `--verbose` mode

Screenshot 2019-07-20 at 01 21 02

**Note:** @johanneswuerbach Skipped few tests which require credential checks.

### CODE REVIEW

File: __tests__/sauce-connect-launcher.test.js [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+            expect(body.status).toBe("terminated");
+            done();
+          });
+            // }, 5000);
+        });
+      });
+    });
+  });
+
+  it.skip("allows to spawn sc detached",  (done) => {
```
Comment (johanneswuerbach) [inline __tests__/sauce-connect-launcher.test.js]: Why is this skipped? CI should be able to run (and pass) that tests, doesn't it?

Comment (valishah) [reply __tests__/sauce-connect-launcher.test.js]: Hello @johanneswuerbach: CI tests are failing and longer to execute, I'l optimize those tests and push back.

Comment (valishah) [reply __tests__/sauce-connect-launcher.test.js]: HI @johanneswuerbach: That test will no longer be valid as saucelabs is not supporting **v4.3.16**

Here are the logs I get while executing that test.  Let me know your thoughts

Screenshot 2019-07-24 at 21 57 00

Comment (valishah) [reply __tests__/sauce-connect-launcher.test.js]: Hi @johanneswuerbach: Converting everything to `mocha` made it a bit tough, So kept the mocha tests along with including jest. We can convert that progressively.

**Comment by valishah** [comment]
@johanneswuerbach: Once this PR is merged, I would like to open a couple more to unify the flow

**Pull request merged**